runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/client/grpc/GrpcWindmillStreamFactory.java

```diff
@@ -209,11 +261,29 @@ public GetWorkerMetadataStream createGetWorkerMetadataStream(
         onNewWindmillEndpoints);
   }
 
-  private StreamObserverFactory newStreamObserverFactory() {
+  private StreamObserverFactory newBufferringStreamObserverFactory() {
     return StreamObserverFactory.direct(
         DEFAULT_STREAM_RPC_DEADLINE_SECONDS * 2, windmillMessagesBetweenIsReadyChecks);
   }
 
+  /**
+   * Simple {@link StreamObserverFactory} that does not buffer or provide extra functionality for
+   * request observers.
+   *
+   * @implNote Used to create stream observers for direct path streams that do not share any
+   *     underlying resources between threads.
+   */
+  private StreamObserverFactory newSimpleStreamObserverFactory() {
```

Comment:
I think that we may still want the isReady checking, as otherwise if outgoing sending is falling behind it just buffers more and more in grpc output stream buffer and can oom.

Reply:
gotcha we may want to bump the deadline then
```
 "Exceeded timeout waiting for the outboundObserver to become ready meaning "
                  + "that the stream deadline was not respected."
```
was getting a lot of these errors in testing.
reverted